src/common/system_memory.h

```diff
@@ -0,0 +1,61 @@
+/*
+
+
+ Copyright 2023 Leil Storage OÜ
+
+ This file is part of SaunaFS.
+
+ SaunaFS is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, version 3.
+
+ SaunaFS is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with SaunaFS. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "common/platform.h"
+
+#include <iostream>
+
+#ifdef _WIN32
+#include <windows.h>
+#elif __linux__
+#include <sys/sysinfo.h>
+#endif
+
+unsigned long long getAvailableMemory() {
+#ifdef _WIN32
+    MEMORYSTATUSEX memInfo;
+    memInfo.dwLength = sizeof(MEMORYSTATUSEX);
+    GlobalMemoryStatusEx(&memInfo);
+    return memInfo.ullAvailPhys;
+#elif __linux__
+    struct sysinfo memInfo;
+    sysinfo(&memInfo);
+    return (memInfo.freeram * memInfo.mem_unit);
+#else
+    return 0; // Unsupported platform
+#endif
+}
+
+unsigned long long getTotalMemory() {
+#ifdef _WIN32
+    MEMORYSTATUSEX memInfo;
+    memInfo.dwLength = sizeof(MEMORYSTATUSEX);
+    GlobalMemoryStatusEx(&memInfo);
+    return memInfo.ullTotalPhys;
+#elif __linux__
+    struct sysinfo memInfo;
+    sysinfo(&memInfo);
+    return (memInfo.totalram * memInfo.mem_unit);
+#else
+    return 0; // Unsupported platform
+#endif
+}
```

Comment:
Maybe create an [interface](https://stackoverflow.com/questions/1216750/how-can-i-simulate-interfaces-in-c) for these functions, so that both windows/linux client (and maybe in the future other clients) can implement this how they like, instead of using ifdefs

Reply:
Done, please check the interface implementation on src/mount/memory_info.h and src/mount/linux_memory_info.h. Review if it is ok that way.